src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -41,7 +41,7 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
             openPolicyAccountingPage(props.policy.id);
         }, [props.policy?.id, isConnectionDataFetchNeeded]);
 
-        if (props.policy?.areConnectionsEnabled && status === 'loading') {
+        if (props.policy?.areConnectionsEnabled && status === 'loading' && shouldBlockView) {
```

Comment:
Remove the status, I think it does not do match, instead use `isConnectionDataFetchNeeded`